com.ibm.wala.util/src/com/ibm/wala/util/intset/IntSetUtil.java

```diff
@@ -31,7 +31,7 @@
       try {
         @SuppressWarnings("unchecked")
         Class<? extends MutableIntSetFactory<?>> intSetFactoryClass = (Class<? extends MutableIntSetFactory<?>>) Class.forName(System.getProperty(INT_SET_FACTORY_CONFIG_PROPERTY_NAME));
-        MutableIntSetFactory<?> intSetFactory = intSetFactoryClass.newInstance();
+        MutableIntSetFactory<?> intSetFactory = intSetFactoryClass.getDeclaredConstructor().newInstance();
```

Comment:
FYI I think I tried this change before but it caused problems for running WALA in a browser /cc @juliandolby

Reply:
Manu is right.  That method broke TeaVM last time I tried it.

Reply:
Indeed, @juliandolby did have to work around this TeaVM problem once before: 2c5d6ed15fa8313759f47a39a1e8a9bf43d0000c. I have reproduced this failure in the [TeaVM sandbox](http://teavm.org/sandbox/index.html) and [reported it to the TeaVM team](https://github.com/konsoletyper/teavm/issues/344) in my usual level of detail.

So hopefully TeaVM will fix this eventually. Until then, I will revise my pull request to exclude the `Class.getDeclaredConstructor()` changes. That is easily to do, as those changes were already encapsulated in their own commit.

Reply:
Is this an issue at execution time on TeaVM or classloading time?  If it's at execution time, we could just basically not use this feature on TeaVM and never execute the `getDeclaredConstructor()` call.  If it's a classloading-time issue then we're stuck.

Reply:
I don’t know when TeaVM would trigger this; perhaps @juliandolby knows.

Either way, let’s keep in mind that the deprecated `Class.newInstance()` method is still available for use in [Java 9](https://docs.oracle.com/javase/9/docs/api/java/lang/Class.html#newInstance--) and even [Java 10](https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#newInstance--). And WALA is stuck at Java 1.8 for the time being anyway. So leaving these calls unchanged for now is perfectly reasonable. If `Class.newInstance()` goes away in Java 11 or later, the TeaVM maintainer (@konsoletyper) has offered [some suggested workarounds](https://github.com/konsoletyper/teavm/issues/344#issuecomment-394842552).